extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/OneTimeAuthTokenRequestHandler.java

```diff
@@ -0,0 +1,197 @@
+package io.quarkus.vertx.http.runtime.security;
+
+import static io.quarkus.security.spi.runtime.SecurityEventHelper.fire;
+import static io.quarkus.vertx.http.runtime.FormAuthRuntimeConfig.CookieSameSite.STRICT;
+import static io.quarkus.vertx.http.runtime.security.FormAuthenticationEvent.*;
+import static io.quarkus.vertx.http.runtime.security.FormAuthenticationMechanism.sendRedirect;
+import static io.quarkus.vertx.http.runtime.security.FormAuthenticationMechanism.startWithSlash;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import jakarta.enterprise.event.Event;
+import jakarta.enterprise.inject.spi.BeanManager;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.runtime.util.HashUtil;
+import io.quarkus.security.AuthenticationFailedException;
+import io.quarkus.security.credential.PasswordCredential;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.security.spi.runtime.SecurityEventHelper;
+import io.quarkus.vertx.http.runtime.FormAuthRuntimeConfig;
+import io.quarkus.vertx.http.security.token.OneTimeAuthenticationTokenSender;
+import io.smallrye.mutiny.Uni;
+import io.vertx.ext.web.RoutingContext;
+
+final class OneTimeAuthTokenRequestHandler {
```

Comment:
```suggestion
final class FormAuthenticationTokenRequestHandler {
```
or, `FormAthenticationTokenHandler`